### PR TITLE
provisor: Error out on unknown server

### PR DESCRIPTION
**Pull request by KellerFuchs**
Required to deal with https://github.com/hashbang/hashbang.sh/issues/58

### CODE REVIEW

File: provisor/provisor.py [hunk 1]
```diff
@@ -9,6 +9,11 @@
 from utils import make_salt, drop_privileges
 from random import shuffle
 
+# An exception class for erroring-out
+#  on unknown hosts
+class UNKNOWN_HOST(Exception):
```
Comment (KellerFuchs) [inline provisor/provisor.py]: If we want to implement more kinds of errors, we probably ought to create a superclass for provisor's errors (and give it a method to return a `{"message": foo}` dictionary?)

**Comment by ChickenNuggers** [comment]
For now, I was assuming we were just going to make it return 400 and the generic error. I don't know how the other admins would want to implement this. Somewhere in our roadmap we had planned to make more descriptive error messages; this could be a good start. I'll view it more later.

**Comment by KellerFuchs** [comment]
@ChickenNuggers Ping for review?

**Comment by ChickenNuggers** [comment]
It looks fine to me. Has this been tested yet?

**Comment by KellerFuchs** [comment]
Not yet, I still haven't figured out how to setup docker for my requirements, nor how to spin up a test instance of the LDAP & website.

**Comment by ChickenNuggers** [comment]
I'm not too sure about test instances as that would require some form of alternate server, which we'd have to pay for out-of-pocket. The Docker command should be in the repo in the systemd service file? The documentation over Docker is simple enough, unless you mean setting up the Docker daemon?

**Comment by KellerFuchs** [comment]
> setup docker for my requirements

I meant setting it up so that it only talks to DockerHub and so on over Tor.

> not too sure about test instances

I meant running the same containers as for the LDAP server and the website (albeit with my patches) on my local machine (not on Digital Ocean), and checking that it works there before deploying to production.

**Comment by ChickenNuggers** [comment]
Not sure about making sure things work over Tor, you should ask #docker in Freenode about that. As for LDAP, I would hook up your machine to talk to the system so that we make sure it works with our setup in case you have a different version or something at your own setup.

**Comment by KellerFuchs** [comment]
> for LDAP, I would hook up your machine to talk to the system

I was talking about spinning up another instance of [the LDAP container](https://github.com/hashbang/docker-slapd) locally, so we know it will be the same software and schema.
In general, I would not want to hook (containers on) my local machine to our LDAP setup, because:
- screwing up badly enough in the thing we are testing might lead to screwing up the live LDAP;
- from a security perspective, it sounds delicate (and LDAP is a sensitive service).

**Comment by ChickenNuggers** [comment]
I guess you're right. I'm still kinda stuck in my stupid "We can always restore from the restore point" thought. I forgot Docker Hub has the images thing. :+1:

**Comment by ChickenNuggers** [comment]
@lrvick - You merged the PR for hashbang.sh that handles the error. Is there a reason you didn't merge this one?

**Comment by KellerFuchs** [comment]
@lrvick Ping.  This is still unmerged, while you merged the matching PR in [hashbang.sh](https://github.com/hashbang/hashbang.sh).

**Comment by KellerFuchs** [comment]
:tableflip:

**Pull request merged**